lambdas/data-migration2/tests/test-granulesAndFiles.js

```diff
@@ -0,0 +1,534 @@
+const Execution = require('@cumulus/api/models/executions');
+const Granule = require('@cumulus/api/models/granules');
+const s3Utils = require('@cumulus/aws-client/S3');
+
+const { dynamodbDocClient } = require('@cumulus/aws-client/services');
+const { fakeFileFactory } = require('@cumulus/api/lib/testUtils');
+const {
+  CollectionPgModel,
+  destroyLocalTestDb,
+  ExecutionPgModel,
+  fakeCollectionRecordFactory,
+  fakeExecutionRecordFactory,
+  FilePgModel,
+  generateLocalTestDb,
+  GranulePgModel,
+  tableNames,
+} = require('@cumulus/db');
+const { RecordAlreadyMigrated } = require('@cumulus/errors');
+
+// eslint-disable-next-line node/no-unpublished-require
+const { migrationDir } = require('../../db-migration');
+const { migrateGranuleRecord, migrateFileRecord, migrateGranulesAndFiles } = require('../dist/lambda/granulesAndFiles');
+
+const buildCollectionId = (name, version) => `${name}___${version}`;
+
+const dateString = new Date().toString();
+const bucket = cryptoRandomString({ length: 10 });
+
+const fileOmitList = ['granule_cumulus_id', 'cumulus_id', 'created_at', 'updated_at'];
+const fakeFile = (alternateBucket) => fakeFileFactory({
+  bucket: alternateBucket || bucket,
+  key: cryptoRandomString({ length: 10 }),
+  size: 1098034,
+  fileName: 'MOD09GQ.A4369670.7bAGCH.006.0739896140643.hdf',
+  checksum: 'checkSum01',
+  checksumType: 'md5',
+  type: 'data',
+  source: 'source',
+});
+
+const generateTestGranule = (collection, executionId, alternateBucket, pdrName, provider) => ({
+  granuleId: cryptoRandomString({ length: 5 }),
+  collectionId: buildCollectionId(collection.name, collection.version),
+  pdrName: pdrName,
+  provider: provider,
+  status: 'running',
+  execution: executionId,
+  cmrLink: cryptoRandomString({ length: 10 }),
+  published: false,
+  duration: 10,
+  files: [
+    fakeFile(alternateBucket),
+  ],
+  error: {},
+  productVolume: 1119742,
+  timeToPreprocess: 0,
+  beginningDateTime: dateString,
+  endingDateTime: dateString,
+  processingStartDateTime: dateString,
+  processingEndDateTime: dateString,
+  lastUpdateDateTime: dateString,
+  timeToArchive: 0,
+  productionDateTime: dateString,
+  timestamp: Date.now(),
+  createdAt: Date.now() - 200 * 1000,
+  updatedAt: Date.now(),
+});
+
+let collectionsModel;
+let executionsModel;
+let granulesModel;
+
+const testDbName = `data_migration_2_${cryptoRandomString({ length: 10 })}`;
+
+test.before(async (t) => {
+  await s3Utils.createBucket(bucket);
+  process.env.stackName = cryptoRandomString({ length: 10 });
+  process.env.system_bucket = bucket;
+
+  process.env.CollectionsTable = cryptoRandomString({ length: 10 });
+  process.env.ExecutionsTable = cryptoRandomString({ length: 10 });
+  process.env.GranulesTable = cryptoRandomString({ length: 10 });
+
+  collectionsModel = new Collection();
+  await collectionsModel.createTable();
+
+  executionsModel = new Execution();
+  await executionsModel.createTable();
```

Comment:
Do we need to create the collection and executions Dynamo tables?

Reply:
Turns out we don't

Reply:
[`92f18be` (#2051)](https://github.com/nasa/cumulus/pull/2051/commits/92f18be28ea3d207ee6021b5bbf883da51b5cbba)